freqtrade/rpc/telegram.py

```diff
@@ -221,25 +220,48 @@ def _format_buy_msg(self, msg: Dict[str, Any]) -> str:
             msg['stake_amount_fiat'] = 0
 
         content = []
-        content.append(
-            f"\N{LARGE BLUE CIRCLE} *{msg['exchange']}:* Buying {msg['pair']}"
-            f" (#{msg['trade_id']})\n"
-        )
-        if msg.get('buy_tag', None):
-            content.append(f"*Buy Tag:* `{msg['buy_tag']}`\n")
-        content.append(f"*Amount:* `{msg['amount']:.8f}`\n")
-        content.append(f"*Open Rate:* `{msg['limit']:.8f}`\n")
-        content.append(f"*Current Rate:* `{msg['current_rate']:.8f}`\n")
-        content.append(
-            f"*Total:* `({round_coin_value(msg['stake_amount'], msg['stake_currency'])}"
-        )
-        if msg.get('fiat_currency', None):
+        # Buy Order Fill
+        if msg['type'] == RPCMessageType.BUY_FILL:
+            content.append(
+                f"\N{CHECK MARK} *{msg['exchange']}:* Bought {msg['pair']}"
+                f" (#{msg['trade_id']})\n"
+            )
+            if msg.get('buy_tag', None):
+                content.append(f"*Buy Tag:* `{msg['buy_tag']}`\n")
+            content.append(f"*Amount:* `{msg['amount']:.8f}`\n")
+            content.append(f"*Open Rate:* `{msg['open_rate']:.8f}`\n")
+            content.append(
+                f"*Total:* `({round_coin_value(msg['stake_amount'], msg['stake_currency'])}"
+            )
+            if msg.get('fiat_currency', None):
+                content.append(
+                    f", {round_coin_value(msg['stake_amount_fiat'], msg['fiat_currency'])}"
+                )
+
+            message = ''.join(content)
+            message += ")`"
```

Comment:
You should be able to move most of these lines out of the if/else branch (most are valid for both buy and buy_fill).
In my understanding, that should be possible for all lines except for the first (where you wanna differ between buy and buy fill) - and maybe one additional line which shows the "actual close" price/rate/profit...

Reply:
Not entirely sure I'm following that feedback, but consolidated it some more so hopefully thats ultimately what you were getting at

Reply:
Yes it is - the benefit of this is - should we decide to change some wording in the future (or add aditional information) - it won't be missed to have it in both versions (if necessary) ...